src/backend/distributed/metadata/node_metadata.c

```diff
@@ -746,8 +746,6 @@ PgDistTableMetadataSyncCommandList(void)
 	metadataSnapshotCommandList = list_concat(metadataSnapshotCommandList,
 											  colocationGroupSyncCommandList);
 
-	/* As the last step, propagate the pg_dist_object entities */
-	Assert(ShouldPropagate());
```

Comment:
Removing the assertion makes sense, but I couldn't see any check with `ShouldPropagate` around here. Are we going to propagate objects to workers even when `citus.enable_ddl_propagation` is set to false?

Reply:
> citus.enable_ddl_propagation

I guess you meant enable_metadata_sync?

> , but I couldn't see any check with ShouldPropagate around here.

I think the assert doesn't make sense, especially inside a function where we create the task list `PgDistTableMetadataSyncCommandList ()`. We can call this function and only print the returning list. Why do we assert `ShouldPropagate` seems like a oversight.

In general, we seem to follow:

```C
if (ThereAreNodesToSync)
{ 
    commands = f();
    .. sync commands
}
```

which makes sense. And, in such structure, `assert` looks redundant (as we  are already inside `if`).

In this case, the caller of `SyncPgDistTableMetadataToNodeList` already filters out the nodes that we should be syncing the metadata. 

One further change I considered was to add `if (!EnableMetadataSync) return` to `SendMetadataCommandListXXX()` function family. But, that is not something I want to do right before the code-freeze.